### DISCUSSION TITLE
Quarkus Kafka Streams Extension with Apicurio Registry and Avro SerDes

### DISCUSSION THREAD

**Discussion by ngc4579** [quarkusio/quarkus#38854] [Q&A]
Hi,

how are the `quarkus.kafka-streams.schema-registry-key` and `quarkus.kafka-streams.schema-registry-url` configuration properties supposed to work in conjunction with a Quarkus Kafka Streams application and Avro SerDes (using an Apicurio registry)?

Specifically: when building a `Topology`, how do I construct the necessary Avro SerDes? Do I have to specifiy the registry URL while configuring the SerDes in code, or is there some 'magic' behind the scenes that allows me to use Avro SerDes while using the configuration properties mentioned above?

This doesn't work:

```
    @Produces
    public Topology buildTopology() {

        final StreamsBuilder streamsBuilder = new StreamsBuilder();

        Serde<Long> agencyKeySerde = new AvroSerde<>(Long.class);
        Serde<Agency> agencySerde = new AvroSerde<>(Agency.class);
        ...
```

When running this with the following configuration:

```
quarkus.kafka-streams.schema-registry-key=apicurio.registry.url
quarkus.kafka-streams.schema-registry-url=http://<redacted>
```

... the application dies at startup complaining that no `apicurio.registry.url` is defined.

[a comment by an automated account is not shown]

**Comment by ngc4579** [comment]
It's been quite a while, and actually I'd like to bring this up again.

Can anyone perhaps share a working example of how to integrate Quarkus Kafka Streams with an Apicurio Registry and Avro Serdes, ideally with auto-registration in order to avoid having to define schema artifacts beforehand?

**Reply by krickert** [reply]
I'm trying to do this right now. I'm in classpath quarkus hell.  You got any luck?

**Reply by ngc4579** [reply]
@krickert No, I've since given up and resorted to using JSON Serdes for the time being...

**Reply by krickert** [reply]
Don't give up. Json is bad. I got mine working. If you want more help lmk

**Reply by ngc4579** [reply]
@krickert Yeah sure - how did you get that working eventually?

**Reply by krickert** [reply]
https://github.com/ai-pipestream/quarkus-apicurio-registry-protobuf

I made a plug-in. Check the integration test.
